cmake/modules/hunter_experimental_add_host_subdirectory.cmake

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2016 Fabian Winnen
+# All rights reserved.
+
+# Adds a subdirectory with CMakeList.txt as seperate build with default (host) toolchain
+
+# Usage:
+#   hunter_add_host_subdirectory(host)
+#   hunter_add_host_subdirectory(host
+#       BINARY_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/host-build
+#       ENV_VARS CC CXX
```

Comment:
> I do not fully understand whats causing the problem

I hope it is clear now (?) Toolchain set CMAKE_CXX_COMPILER and CMake do set `CC` environment variable using value of CMAKE_CXX_COMPILER. When we do run `execute_process` environment variable `CC` inherited and used as a compiler (CMake read `CC` environment variable if it's set).

> Do you mean including clear-all.cmake on client side in host/CMakeList.txt before project

Well in this case there must be no problems with using `clear-all.cmake`.

> in hunter_experimental_add_host_subdirectory before execute_process

Yes I mean this approach. Can you please try it and see what happens?

> The Problem I see with the later is, that it may clear some variables purposefully set by some toolchain and not restore them afterwards

We should not use those variables in build. Environment variables customization doesn't scale well. May be it shows us some hidden problems.

> But I don't understand what you mean with

I mean something like this:

```cmake
file(READ "${_args_BINARY_DIRECTORY}/_3rdParty/Hunter/install-root-dir" HUNTER_HOST_ROOT)
set(HUNTER_HOST_ROOT "${HUNTER_HOST_ROOT}" PARENT_SCOPE)
```

so we can use `HUNTER_HOST_ROOT` in target's CMakeLists.txt without reading Hunter internal stuff:

```
hunter_add_host_project(host) # set HUNTER_HOST_ROOT variable
add_executable(protobuf::protoc IMPORTED)
set_property(TARGET protobuf::protoc APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(protobuf::protoc PROPERTIES IMPORTED_LOCATION_RELEASE "${HUNTER_HOST_ROOT}/bin/protoc")
```